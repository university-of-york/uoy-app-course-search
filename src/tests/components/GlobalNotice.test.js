import { render, screen } from "@testing-library/react";
import { GlobalNotice } from "../../components/GlobalNotice.js";

describe("GloablNotice", () => {
    it("renders child content as expected", () => {
        const children = <h2>This is a notice</h2>;

        const { asFragment } = render(<GlobalNotice>{children}</GlobalNotice>);

        expect(asFragment()).toMatchSnapshot();
    });

    it("does not render any content if no children are present", () => {
        render(<GlobalNotice />);

        expect(screen.queryByText(/./)).not.toBeInTheDocument();
    });
});
