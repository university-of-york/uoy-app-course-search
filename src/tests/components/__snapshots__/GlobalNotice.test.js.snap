// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalNotice renders child content as expected 1`] = `
<DocumentFragment>
  <div
    class="c-global-notice c-global-notice--highlighted"
  >
    <h2>
      This is a notice
    </h2>
  </div>
</DocumentFragment>
`;
